test_conformance/subgroups/subhelpers.h

```diff
@@ -1444,6 +1449,35 @@ template <typename Ty, typename Fns> class KernelExecutor {
 
         return status;
     }
+
+public:
+    test_status run_and_check(WorkGroupParams &test_params)
+    {
+        test_status tmp_status = TEST_SKIPPED_ITSELF;
+
+        if (test_params.cluster_size_arg != -1)
+        {
+            for (cl_uint cluster_size = 1;
+                 cluster_size <= test_params.subgroup_size; cluster_size *= 2)
+            {
+                test_params.cluster_size = cluster_size;
+                cl_int error =
+                    clSetKernelArg(kernel, test_params.cluster_size_arg,
+                                   sizeof(cl_uint), &cluster_size);
+                test_error_fail(error, "Unable to set cluster size");
+
+                tmp_status = run_and_check_with_cluster_size(test_params);
+
+                if (tmp_status == TEST_FAIL) break;
+            }
+        }
+        else
+        {
+            tmp_status = run_and_check_with_cluster_size(test_params);
+        }
```

Comment:
> Mostly LGTM, I left one comment in an area where I was a bit confused. I've also asked @gwawiork to do a review.
> 
> Do I understand correctly that this is an infrastructure-only change? It doesn't look like the new "cluster size arg" is being set anywhere (yet), so the test coverage is currently the same before and after this change.

Yes, that's correct - as the description states: "This will be used for the testing of cl_khr_subgroup_clustered_reduce." - so it's not used yet but it will be in a later change for cl_khr_subgroup_clustered_reduce when that change is ready. I didn't want to delay the review of those subsequent changes, which is why I submitted this PR with just the infrastructure change.

> I was a little confused by this initially, because this case where the cluster_size_arg is equal to -1 (and hence irrelevant) is still calling run_and_check_with_cluster_size. I'm not exactly sure how to fix this, or even if this is a problem that needs to be solved, but one easy option would be to rename run_and_check_with_cluster_size to something like run_and_check_helper, so at least it's not so explicitly coupled to the cluster size?

Initially I called this `do_run_and_check` and was advised internally that this wasn't sufficiently distinct from `run_and_check`. I'd be happy to rename this in whatever way is deemed acceptable.

Reply:
Thinking out loud: Another option we could consider is to keep the previous function named as it was before (`run_and_check`) and then choose a more descriptive name for the new function, perhaps something like `run_and_check_all_cluster_sizes`.

Reply:
I feel that this thread may be walking into bikeshedding territory :)

The function in question is private to the class, and thus not going to be used elsewhere. Perhaps the current name is sufficient?